ibm/resource_ibm_appid_application_scopes_test.go

```diff
@@ -0,0 +1,78 @@
+package ibm
+
+import (
+	"fmt"
+	appid "github.com/IBM/appid-management-go-sdk/appidmanagementv4"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"strings"
+	"testing"
+)
+
+func TestAccIBMAppIDApplicationScopes_basic(t *testing.T) {
+	appName := fmt.Sprintf("tf_testacc_app_scopes_%d", acctest.RandIntRange(10, 100))
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckIBMAppIDApplicationScopesDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckIBMAppIDApplicationScopesConfig(appIDTenantID, appName),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("ibm_appid_application_scopes.scopes", "scopes.#", "3"),
+					resource.TestCheckResourceAttr("ibm_appid_application_scopes.scopes", "scopes.0", "scope1"),
+					resource.TestCheckResourceAttr("ibm_appid_application_scopes.scopes", "scopes.1", "scope2"),
+					resource.TestCheckResourceAttr("ibm_appid_application_scopes.scopes", "scopes.2", "scope3"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckIBMAppIDApplicationScopesConfig(tenantID string, name string) string {
+	return fmt.Sprintf(`
+		resource "ibm_appid_application" "test_app" {
+			tenant_id = "%s"
+			name = "%s"
+		}
+
+		resource "ibm_appid_application_scopes" "scopes" {
+		  tenant_id = ibm_appid_application.test_app.tenant_id
+		  client_id = ibm_appid_application.test_app.client_id
+		  scopes = ["scope1", "scope2", "scope3"]
+		}
+	`, tenantID, name)
+}
+
+func testAccCheckIBMAppIDApplicationScopesDestroy(s *terraform.State) error {
+	appIDClient, err := testAccProvider.Meta().(ClientSession).AppIDAPI()
+
+	if err != nil {
+		return err
+	}
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "ibm_appid_application_scopes" {
+			continue
+		}
+
+		id := rs.Primary.ID
+		idParts := strings.Split(id, "/")
+
+		tenantID := idParts[0]
+		clientID := idParts[1]
+
+		_, _, err := appIDClient.GetApplicationScopes(&appid.GetApplicationScopesOptions{
```

Comment:
This works, because we have application resource which gets destroyed. Technically it should return an empty scopes array, but I'm not sure if there is a way to test it before application resource is gone. Same thing for `application_roles`